### DISCUSSION TITLE
Would you maybe look into supporting rustic too?

### DISCUSSION THREAD

**Discussion by ovizii** [garethgeorge/backrest#117] [Ideas]
Here is a short comparison: https://rustic.cli.rs/docs/comparison-restic.html I'm particularly fond of the decrease in memory requirement and the lock-free operations. => https://rustic.cli.rs/docs/comparison-restic.html

**Comment by garethgeorge** [comment]
Rustic support is probably not in scope at the moment but I'm definitely keeping an eye on the project. I'd like to see it grow a bit more mature. 

To support rustic I'd also need bandwidth to convert from it's data model (which is a programatic API rather than JSON based) to a representation compatible with baseline restic which should be very doable but adds to maintenance burden. For the time being I'm focused on getting backrest to support all of restic's base feature set -- and doing so reliably!

**Comment by ovizii** [comment]
Great, it is good to know you've spotted rustic and possibly keeping it in mind for future support. Thanks for your great work with this project.

**Comment by alexandru0-dev** [comment]
rustic support would be fantastic, but at the same time rustic lacks full tests (at least at time of writing)a and also backrest lacks full support of rustic features.
also idk how easy would be to integrate rustic as it's not 100% compatible with the restic cli
@garethgeorge if in supporting it, you find some blocking issues pls create an issue on the rustic repo :D
thx for the great work

**Comment by nickchomey** [comment]
I'd also love to see this - rustic is far faster and more featureful than restic. The devs are very responsive and active too, while the restic devs seem to have been... resting... for the past couple years

**Reply by garethgeorge** [reply]
Agreed that rustic support would be neat -- I have a few open issues with the rustic devs where I've gotten a bit of input on what would be needed for Backrest to support rustic.

The biggest thing that's been a blocker so far is that, while rustic does offer a programatic API, that API is quite low level. On the flip side of things, rustic's CLI is very high level and seems to mostly do plaintext output and uses a centralized config (which is not compatible with Backrest).

So to support rustic, I need to create a rust binary that wraps rustic's pkg with a simple gRPC API or similar.

**Reply by nickchomey** [reply]
Thanks for the response and glad to hear it is on your radar!

This is all beyond my capabilities, so I don't have any suggestions other than to trying following up with them for assistance with making rustic more accessible to you. They really are quite responsive.

**Comment by Quiwy** [comment]
Hi @garethgeorge, thank you for your great work! 😊

I just discovered Backrest on the [Yunohost forum](https://forum.yunohost.org/t/backrest-web-ui-and-orchestrator-for-restic-backup), and it looks very promising. The only thing holding me back at the moment is the lack of support for Glacier storage, which seems to be supported by Rustic.

If Backrest could support Rustic, it would be amazing to have Glacier storage support in Backrest as well. I'll keep an eye on this discussion!
